subprojects/docs/docs.gradle

```diff
@@ -415,6 +416,15 @@ samples {
     }
 }
 
+// The org.gradle.samples plugin uses Exemplar to execute integration tests on the samples.
+// Exemplar doesn't know about that it's running in the context of the gradle/gradle build
+// so it uses the Gradle distribution from the running build. This is not correct, because
+// we want to verify that the samples work with the Gradle distribution being built.
+tasks.named('docsTest').configure() { task ->
+    task.dependsOn(tasks.named('intTestImage'))
+    task.systemProperty("integTest.gradleHomeDir", new File(project.buildDir, "integ test").absolutePath)
+}
```

Comment:
Is there some way we can reuse `GradleInstallationForTestEnvironmentProvider` in DistributionTest.kt so that docsTest could be made cacheable too?

Reply:
I've added a todo for that. By the looks of it, we'd need to move the test task definition and the GradleInstallationForTestEnvironmentProvider to the same place to be able to do that.